lib/mpl_toolkits/axisartist/axis_artist.py

```diff
@@ -926,10 +944,17 @@ def _update_ticks(self, renderer=None):
             renderer = self.get_figure(root=True)._get_renderer()
 
         dpi_cor = renderer.points_to_pixels(1.)
-        if self.major_ticks.get_visible() and self.major_ticks.get_tick_out():
+        if self.major_ticks.get_visible():
             ticklabel_pad = self.major_ticks._ticksize * dpi_cor
-            self.major_ticklabels._external_pad = ticklabel_pad
-            self.minor_ticklabels._external_pad = ticklabel_pad
+            if self.major_ticks.get_tick_direction() == "in":
+                self.major_ticklabels._external_pad = 0
+                self.minor_ticklabels._external_pad = 0
+            elif self.major_ticks.get_tick_direction() == "out":
+                self.major_ticklabels._external_pad = ticklabel_pad
+                self.minor_ticklabels._external_pad = ticklabel_pad
+            else:  # inout
+                self.major_ticklabels._external_pad = ticklabel_pad / 2
+                self.minor_ticklabels._external_pad = ticklabel_pad / 2
```

Comment:
Optional: Is this more readable?
```suggestion
            ticksize = self.major_ticks._ticksize * dpi_cor
            ticklabel_pad = {
                "in": 0,
                "out": ticksize,
                "inout": ticksize / 2
            }[major_ticks.get_tick_direction()]
            self.major_ticklabels._external_pad = ticklabel_pad
            self.minor_ticklabels._external_pad = ticklabel_pad
```
Note: This would also replace the previous line `ticklabel_pad = self.major_ticks._ticksize * dpi_cor`, but GitHub doesn't allow me to make a suggestion with including that line ("Applying suggestion on deleted lines is not possible")

Reply:
Sure, did something along these lines.